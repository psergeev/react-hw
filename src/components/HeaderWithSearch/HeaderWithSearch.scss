header {
    background-color: #222;
    padding: 20px 50px;

    .title {
        color: #b3505d;
        font-weight: bold;
        font-size: 18px;
        margin: 0;
        padding: 0;
    }

    .find-text {
        margin: 40px 0 20px 0;
        text-transform: uppercase;
        font-size: 20px;
        color: white;
    }

    .search-input {
        background-color: black;

        >div {
            width: 100%;
            padding: 5px 0;

            &:before {
                border-bottom: solid #75383e 2px;
            }

            &:hover:before {
                border-bottom: none;
            }
        }

        input {
            font-size: 16px;
            margin: 0 15px;
            color: white;
        }
    }

    .search-controls {
        margin-top: 20px;
        display: flex;
        justify-content: space-between;
        font-size: 12px;

        .search-by {
            >span {
                margin-top: 5px;
            };
        }

        span {
            text-transform: uppercase;
            color: white;
        }

        .searchButton {
            background-color: #F45263;
            text-transform: uppercase;
            padding: 10px 60px;
        }
    }
}
