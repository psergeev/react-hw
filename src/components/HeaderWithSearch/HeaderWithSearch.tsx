import * as React from 'react';
import './HeaderWithSearch.scss';
import SearchByButton from '../SearchByButton';

interface State {
    textTyped: string;
    searchBy: string;
}

export default class extends React.PureComponent<{}, State> {
    public constructor(props: {}) {
        super(props);

        this.state = {
            textTyped: '',
            searchBy: 'title'
        };

        this._handleKeyPress = this._handleKeyPress.bind(this);
        this._handleSearchByClick = this._handleSearchByClick.bind(this);
    }

    private _handleSearchByClick(value: string) {
        this.setState({
            searchBy: value
        });
    }

    private _handleKeyPress(event: React.ChangeEvent<HTMLInputElement>) {
        this.setState({
            textTyped: (event.target as any).value
        });
    }

    public render() {
        return (
            <header>
                <h1 className="title">netflixroulette</h1>
                <p className="find-text">find your movie</p>
                <div className="search-input">
                    <input
                        value={this.state.textTyped}
                        onChange={this._handleKeyPress}
                    />
                </div>
                <div className="search-controls">
                    <div className="search-by">
                        <span>Search by</span>

                        <SearchByButton
                            value="title"
                            selected={this.state.searchBy === 'title'}
                            handleSearchByClick={this._handleSearchByClick}
                        />

                        <SearchByButton
                            value="genre"
                            selected={this.state.searchBy === 'genre'}
                            handleSearchByClick={this._handleSearchByClick}
                        />
                    </div>

                    <button type="button">Search</button>
                </div>
            </header>
        );
    }
}
