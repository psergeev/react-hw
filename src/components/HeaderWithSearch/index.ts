import HeaderWithSearch from './HeaderWithSearch';

export default HeaderWithSearch;
