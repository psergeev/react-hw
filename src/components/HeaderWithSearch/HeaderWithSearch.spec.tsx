import { shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import HeaderWithSearch from './HeaderWithSearch';

jest.mock('../SearchByButton', () => 'SearchByButton');

const props = {
    searchBy: 'title',
    sortBy: 'rating',
    search: '',
    handleSearchByChange: jest.fn(),
    handleSearchClick: jest.fn(),
};

test('HeaderWithSearch component works as expected', () => {
    const component = renderer.create(
        <HeaderWithSearch {...props} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('HeaderWithSearch should switch search by button', () => {
    const component = shallow(<HeaderWithSearch {...props} searchBy="genre" />);

    expect(component).toMatchSnapshot();
});


test('HeaderWithSearch should show typed text', () => {
    const component = shallow(<HeaderWithSearch {...props} />);

    component.setState({
        textTyped: '123'
    });

    expect(component).toMatchSnapshot();
});
