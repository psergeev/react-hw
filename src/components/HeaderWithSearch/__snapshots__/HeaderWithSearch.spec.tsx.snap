// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderWithSearch component works as expected 1`] = `
<header>
  <h1
    className="title"
  >
    netflixroulette
  </h1>
  <p
    className="find-text"
  >
    find your movie
  </p>
  <div
    className="search-input"
  >
    <input
      onChange={[Function]}
      value=""
    />
  </div>
  <div
    className="search-controls"
  >
    <div
      className="search-by"
    >
      <span>
        Search by
      </span>
      <SearchByButton
        handleSearchByClick={[Function]}
        selected={true}
        value="title"
      />
      <SearchByButton
        handleSearchByClick={[Function]}
        selected={false}
        value="genre"
      />
    </div>
    <button
      type="button"
    >
      Search
    </button>
  </div>
</header>
`;

exports[`HeaderWithSearch should show typed text 1`] = `
<header>
  <h1
    className="title"
  >
    netflixroulette
  </h1>
  <p
    className="find-text"
  >
    find your movie
  </p>
  <div
    className="search-input"
  >
    <input
      onChange={[Function]}
      value="123"
    />
  </div>
  <div
    className="search-controls"
  >
    <div
      className="search-by"
    >
      <span>
        Search by
      </span>
      <SearchByButton
        handleSearchByClick={[Function]}
        selected={true}
        value="title"
      />
      <SearchByButton
        handleSearchByClick={[Function]}
        selected={false}
        value="genre"
      />
    </div>
    <button
      type="button"
    >
      Search
    </button>
  </div>
</header>
`;

exports[`HeaderWithSearch should switch search by button 1`] = `
<header>
  <h1
    className="title"
  >
    netflixroulette
  </h1>
  <p
    className="find-text"
  >
    find your movie
  </p>
  <div
    className="search-input"
  >
    <input
      onChange={[Function]}
      value=""
    />
  </div>
  <div
    className="search-controls"
  >
    <div
      className="search-by"
    >
      <span>
        Search by
      </span>
      <SearchByButton
        handleSearchByClick={[Function]}
        selected={false}
        value="title"
      />
      <SearchByButton
        handleSearchByClick={[Function]}
        selected={true}
        value="genre"
      />
    </div>
    <button
      type="button"
    >
      Search
    </button>
  </div>
</header>
`;
